src/common/container/lock_free_array.cpp

```diff
@@ -53,6 +53,7 @@ template <typename ValueType>
 void LOCK_FREE_ARRAY_TYPE::Erase(const std::size_t &offset,
                                  const ValueType &invalid_value) {
   LOG_TRACE("Erase at %lu", offset);
+  PELOTON_ASSERT(lock_free_array.size() > offset);
```

Comment:
Again, was this for debugging? Or have you introduced some new logic that could cause this condition?

Reply:
No new logic. .at() will throw an exception that we never catch if it's out of bounds. Seems more correct to me to have a PELOTON_ASSERT first. Also, see Find() and Update() which already have that logic. Erase() not having it just looks like an oversight.